packages/next/build/utils.ts

```diff
@@ -126,227 +134,269 @@ export async function printTreeView(
       // Remove file hash
       .replace(/(?:^|[.-])([0-9a-z]{6})[0-9a-z]{14}(?=\.)/, '.$1')
 
-  const messages: [string, string, string][] = [
-    ['Page', 'Size', 'First Load JS'].map((entry) =>
-      chalk.underline(entry)
-    ) as [string, string, string],
-  ]
-
+  // Check if we have a custom app.
   const hasCustomApp = await findPageFile(pagesDir, '/_app', pageExtensions)
-  pageInfos.set('/404', {
-    ...(pageInfos.get('/404') || pageInfos.get('/_error')),
-    static: useStatic404,
-  } as any)
 
-  if (!list.includes('/404')) {
-    list = [...list, '/404']
-  }
+  const filterAndSortList = (list: ReadonlyArray<string>) =>
+    list
+      .slice()
+      .filter(
+        (e) =>
+          !(
+            e === '/_document' ||
+            e === '/_error' ||
+            (!hasCustomApp && e === '/_app')
+          )
+      )
+      .sort((a, b) => a.localeCompare(b))
 
-  const sizeData = await computeFromManifest(
-    buildManifest,
+  // Collect all the symbols we use so we can print the icons out.
+  const usedSymbols = new Set()
+
+  const messages: [string, string, string][] = []
+
+  const stats = await computeFromManifest(
+    { build: buildManifest, app: appBuildManifest },
     distPath,
     gzipSize,
     pageInfos
   )
 
-  const usedSymbols = new Set()
+  const printFileTree = async ({
+    list,
+    routerType,
+  }: {
+    list: ReadonlyArray<string>
+    routerType: ROUTER_TYPE
+  }) => {
+    messages.push(
+      [routerType === 'app' ? 'Route' : 'Page', 'Size', 'First Load JS'].map(
```

Comment:
```suggestion
      [routerType === 'app' ? 'Route' : 'Page', 'Size', 'First Load JS shared by all'].map(
```
Did we mean to drop the `shared by all` here?